cmd/repo.go

```diff
@@ -191,29 +200,38 @@ func scanErcoleReposerviceRepository(upstreamRepo config.UpstreamRepository) (re
 }
 
 // scanRepository scan a single repository and return detected files
-func scanRepository(repo config.UpstreamRepository) (repo.Index, error) {
-	switch repo.Type {
-	case "github-release":
-		return scanGithubReleaseRepository(repo)
-	case "directory":
-		return scanDirectoryRepository(repo)
-	case "ercole-reposervice":
-		return scanErcoleReposerviceRepository(repo)
-	default:
-		return nil, fmt.Errorf("Unknown repository type %q of %q", repo.Type, repo.Name)
+func scanRepository(upstreamRepo config.UpstreamRepository) (repo.Index, error) {
+	if strings.TrimSpace(upstreamRepo.Name) == "local" {
+		return nil,
+			fmt.Errorf("\"local\" isn't a valid name for an upstream repository")
 	}
```

Comment:
Maybe move the check down within the switch?
```
	case repo.UpstreamTypeLocal:
		return nil,
			fmt.Errorf("\"local\" isn't a valid type for an upstream repository")
```

Reply:
If you check it only for UpstreamTypeLocal, you could as well have "local" for a repo you defined and others "local" files for the true ones, added by this PR.
Instead, if you check for every repo, you assure you haven't two repo with the same "local" name, as far as silly this could be.

Reply:
Ok